src/framework/app-base.js

```diff
@@ -2037,6 +2037,7 @@ class AppBase extends EventHandler {
 
         this.tick = null;
 
+        this.fire('destroy', this); // fire destroy event
```

Comment:
I wonder if we should fire this at the start of the `destroy` function instead of here, giving the user an option to use the AppBase instance while handling this event (for example to save some data), instead of at a point where it's mostly not valid anymore.

Reply:
100% agree here.

Reply:
Yep, good point, just moved it up